specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2025-03-03/gallery.json

```diff
@@ -1400,6 +1400,539 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/scripts/{galleryScriptName}": {
+      "put": {
+        "tags": [
+          "GalleryScripts"
+        ],
+        "operationId": "GalleryScripts_CreateOrUpdate",
+        "description": "Create or update a Gallery Script Definition. Gallery scripts allow the storage, sharing and reuse of common scripts",
+        "parameters": [
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9]+([_]?[a-zA-Z0-9]+)*$",
+            "description": "The name of the Gallery in which the Script Definition is to be created."
+          },
+          {
+            "name": "galleryScriptName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9]+([_]?[a-zA-Z0-9]+)*$",
+            "description": "The name of the gallery Script Definition to be created or updated. The allowed characters are alphabets and numbers with dots, dashes, and periods allowed in the middle. The maximum length is 80 characters."
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "galleryScript",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryScript"
+            },
+            "description": "Parameters supplied to the create or update gallery Script operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```

Comment:
>201 [](http://example.com/codeflow?start=11&length=3)

Do you have any polling headers ?

please check [https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-replacing-or-updating-resources-asynchronously](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-replacing-or-updating-resources-asynchronously)